microprofile-metrics/src/test/java/org/jboss/eap/qe/microprofile/metrics/MetricsTest.java

```diff
@@ -0,0 +1,320 @@
+    @Test
+    @RunAsClient
+    public void dataTest() {
+        get(deploymentUrl.toString()).then()
+                .statusCode(200)
+                .body(equalTo("Hello from counted and timed and metered and concurrent-gauged method"));
+
+        given().baseUri(deploymentUrl.toString()).basePath("another-hello").get().then()
+                .statusCode(200)
+                .body(equalTo("Hello from another counted method"));
+
+        for (int i = 0; i < 10; i++) {
+            get(deploymentUrl.toString()).then()
+                    .statusCode(200);
+        }
+
+        jsonData();
+        prometheusData();
+    }
+
+    public void jsonData() {
+        jsonMetricsRequest.get().then()
+                .contentType(ContentType.JSON)
+                .header("Content-Type", containsString("application/json"))
+                .body("$", hasKey("application"),
+                        "application", hasKey("hello-count"),
+                        "application.hello-count", equalTo(12), // 11 at `/` + 1 at `/another-hello`
+
+                        "application", hasKey("hello-time"),
+                        "application.hello-time.count", equalTo(11),
+                        "application.hello-time.min.toDouble()", greaterThanOrEqualTo(1.0),
+                        "application.hello-time.min.toDouble()", lessThanOrEqualTo(110.0),
+                        "application.hello-time.max.toDouble()", greaterThanOrEqualTo(1.0),
+                        "application.hello-time.max.toDouble()", lessThanOrEqualTo(110.0),
+
+                        "application", hasKey("hello-freq"),
+                        "application.hello-freq.count", equalTo(11),
+
+                        "application", hasKey("hello-invocations"),
+                        "application.hello-invocations.current", equalTo(0));
+    }
+
+    public void prometheusData() {
+        textMetricsRequest.get().then()
+                .contentType(ContentType.TEXT)
+                .header("Content-Type", containsString("text/plain"))
+                .body(
+                        containsString("application_hello_count_total 12.0"),
+                        containsString("application_hello_time_seconds_count 11.0"),
+                        containsString("application_hello_time_seconds_count 11.0"),
+                        new PrometheusMetricValueMatcher("application_hello_time_min_seconds", greaterThanOrEqualTo(0.001),
+                                lessThanOrEqualTo(0.110)),
+                        containsString("application_hello_freq_total 11.0"),
+                        containsString("application_hello_invocations_current 0.0"),
+                        new PrometheusMetricValueMatcher("application_hello_time_min_seconds", greaterThanOrEqualTo(0.001),
+                                lessThanOrEqualTo(0.110)));
+    }
+
+    /**
+     * @tpTestDetails Smoke scenario to verify correct behaviour
+     * @tpPassCrit Correct data are provided for metrics
```

Comment:
How it is decided that data are correct?

Reply:
Decision is based on MP Metrics specification.